frontend/src/metabase/static-viz/components/XYChart/Values/Values.tsx

```diff
@@ -0,0 +1,381 @@
+import React from "react";
+import _ from "underscore";
+
+import { Text } from "@visx/text";
+import { scaleBand } from "@visx/scale";
+
+import { getValueStep, getY } from "../utils";
+
+import type { TextProps } from "@visx/text";
+import type { AnyScaleBand, PositionScale } from "@visx/shape/lib/types";
+import type {
+  HydratedSeries,
+  SeriesDatum,
+  StackedDatum,
+  VisualizationType,
+  XYAccessor,
+} from "../types";
+
+const VALUES_MARGIN = 6;
+const VALUES_STROKE_MARGIN = 3;
+const FLIPPED_VALUES_MARGIN = VALUES_MARGIN + 8;
+
+interface ValuesProps {
+  series: HydratedSeries[];
+  formatter: (value: number, compact: boolean) => string;
+  valueProps: Partial<TextProps>;
+  xScale: XScale;
+  yScaleLeft: PositionScale | null;
+  yScaleRight: PositionScale | null;
+  innerWidth: number;
+  areStacked: boolean;
+  xAxisYPos: number;
+}
+
+interface XScale {
```

Comment:
I'd put this type declatation to the file that contains the function that returns it, wdyt?